src/dmd/lexer.d

```diff
@@ -145,6 +145,24 @@ class Lexer
         }
     }
 
+    version (DMDLIB)
```

Comment:
Also, I don't see thee need of being restricted to `DMDLIB`.

Reply:
People might argue that this will increase the size of the binary.

Reply:
The linker has options to strip unused symbols on the final binary, `-Wl,--gc-sections`. But sure, makes sense, in terms of usability in the compiler.